processor/eventfilter/eventfilter.go

```diff
@@ -69,3 +77,325 @@ func GetSupportedMessageEvents(destination *backendconfig.DestinationT) ([]strin
 	}
 	return nil, false
 }
+
+type AllowTransformerEventParams struct {
+	TransformerEvent      *transformer.TransformerEventT
+	SupportedMessageTypes []string
+}
+
+type EventParams struct {
+	MessageType string
+}
+
+type ConnectionModeFilterParams struct {
+	Destination      *backendconfig.DestinationT
+	SrcType          string
+	Event            *EventParams
+	DefaultBehaviour bool
+}
+
+func getMessageType(event *types.SingularEventT) string {
+	eventMessageTypeI := misc.MapLookup(*event, "type")
+	if eventMessageTypeI == nil {
+		pkgLogger.Error("Message type is not being sent for the event")
+		return ""
+	}
+	eventMessageType, isEventTypeString := eventMessageTypeI.(string)
+	if !isEventTypeString {
+		pkgLogger.Errorf("Invalid message type :%v, Type assertion failed", eventMessageTypeI)
+		return ""
+	}
+	return eventMessageType
+}
+
+/*
+This methods lets the caller know if we need to allow the event to proceed to destination transformation
+
+Currently this method supports below validations(executed in the same order)
+
+1. Validate if messageType sent in event is included in SupportedMessageTypes
+
+2. Validate if the event is sendable to destination based on connectionMode, sourceType & messageType
+*/
+func AllowEventToDestTransformation(transformerEvent *transformer.TransformerEventT, supportedMsgTypes []string) (bool, *transformer.TransformerResponseT) {
```

Comment:
[minor] idiomatic function comment,  should begin with function name

https://tip.golang.org/doc/comment

```suggestion
/*
AllowEventToDestTransformation lets the caller know if we need to allow the event to proceed to destination transformation.

Currently this method supports below validations(executed in the same order):
1. Validate if messageType sent in event is included in SupportedMessageTypes
2. Validate if the event is sendable to destination based on connectionMode, sourceType & messageType
*/
func AllowEventToDestTransformation(transformerEvent *transformer.TransformerEventT, supportedMsgTypes []string) (bool, *transformer.TransformerResponseT) {
```

Reply:
Except the spacing, taken care of other things